flink-streaming-java/src/main/java/org/apache/flink/streaming/api/environment/StreamContextEnvironment.java

```diff
@@ -36,28 +31,16 @@
 
 	private static final Logger LOG = LoggerFactory.getLogger(StreamContextEnvironment.class);
 
-	private final List<URL> jars;
-
-	private final List<URL> classpaths;
-	
-	private final Client client;
+	private final ContextEnvironment ctx;
 
 	private final ClassLoader userCodeClassLoader;
-	
-	private final boolean wait;
 
-	protected StreamContextEnvironment(Client client, List<URL> jars, List<URL> classpaths, int parallelism,
-			boolean wait) {
-		this.client = client;
-		this.jars = jars;
-		this.classpaths = classpaths;
-		this.wait = wait;
-		
-		this.userCodeClassLoader = JobWithJars.buildUserCodeClassLoader(jars, classpaths,
-				getClass().getClassLoader());
+	protected StreamContextEnvironment(ContextEnvironment ctx) {
+		this.ctx = ctx;
+		this.userCodeClassLoader = JobWithJars.buildUserCodeClassLoader(ctx.getJars(), ctx.getClasspaths(), getClass().getClassLoader());
```

Comment:
Okay. I will go through all usages of classloaders and try to delegate them to the `Client`.